examples/with-claims/api/claims/RolesClaim.ts

```diff
@@ -0,0 +1,91 @@
+import Session, { SessionClaim, SessionClaimValidator, SessionContainer } from "supertokens-node/recipe/session";
+import { roleDB } from "../mockRoleDb";
+
+class RolesClaimClass extends SessionClaim<string[]> {
+    constructor() {
+        super("st-r");
+        this.hasRole = {
+            validatorTypeId: this.key,
+            including: (role: string) => ({
+                validatorTypeId: this.key,
+                claim: this,
+                shouldRefetch: (payload, ctx) => {
+                    return this.getValueFromPayload(payload, ctx) === undefined;
+                },
+                validate: async (payload, ctx) => {
+                    return {
+                        isValid: this.getValueFromPayload(payload, ctx).includes(role),
+                        reason: { missingRole: role },
```

Comment:
Why do we give a reason if `isValid` is `true`?

Reply:
That's not necessary (and incorrectly typed), but it doesn't cause any errors. I'll update it.